spec/decrypt_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'aes256gcm_decrypt'
+require 'base64'
+require 'digest'
+
+describe 'Aes256GcmDecrypt::decrypt' do
+  before(:each) do
+    @ciphertext_and_tag = Base64.decode64(File.read('spec/token_data_base64.txt'))
+    @key = [File.read('spec/key_hex.txt').strip].pack('H*')
+  end
+
+  it 'decrypts correctly' do
+    plaintext = Aes256GcmDecrypt::decrypt(@ciphertext_and_tag, @key)
+    digest = Digest::SHA256.base64digest(plaintext)
+    expect(digest).to eq '6Nltyg0FgJxJ8wt6D6XDZ0y4iRD53kLoQcTBQ0FEvyY='
+  end
+
+  it 'detects wrong parameter types' do
+    expect{Aes256GcmDecrypt::decrypt(42, @key)}.to raise_error(TypeError)
+    expect{Aes256GcmDecrypt::decrypt(@ciphertext_and_tag, 42)}.to raise_error(TypeError)
+  end
+
+  it 'detects too short ciphertext_and_tag' do
+    (0..16).each do |i|
+      ciphertext_and_tag = 'x' * i
+      expect(Aes256GcmDecrypt::decrypt(ciphertext_and_tag, @key)).to be nil
```

Comment:
No error?

Reply:
I have given it a shot in https://github.com/clearhaus/aes256gcm_decrypt/pull/8/commits/30286d8ef9e47b2556cb3132627621bd4dfb324d :slightly_smiling_face: